### ISSUE TITLE
GTA 4 flickering dots using DXVK 1.9 on AMD RDNA2

### ISSUE THREAD

**Issue by Fezzy976** [doitsujin/dxvk#2179]
###Issue Description
Issues using latest DXVK 1.9 with AMD 6800XT and GTA 4. Producing flickering coloured crosshairs on screen when smoke or bloom effects are used/switched. Flicker happens for a split second so capturing this was tricky.

Drivers used are 20.6.1, 20.6.2, 20.7.1, and UWP drivers. All exhibit the same issue, the issue is not present when using DX. GPU is not overclocked either and no overheating has been monitored. No other game displays this issue.

The issue can be reproduced standing next to a light or smoke source in game and moving camera around. Issue can be minor to severe. If I enabled AMD VSR and run the game at 5K144Hz the flickering is VERY noticeable. Moving resolution down but keeping framerate high does somewhat mitigate the issue but its still there. Capping the fps to 72fps at 2K is where the issue once again becomes more apparent.

No overclocking is enabled on the system either on the GPU or CPU. And temps are well within spec.

###Log File
No log file can be produced as this is not a crash its a graphical issue that is not present in DirectX and only present when running DXVK.

### Software information
GTA IV Complete Edition
2560x1400 144Hz (72fps cap using RTSS)
Max settings
View Distance Slider = 43
Vsync ON

### System information
- GPU: 6800XY (no overclock)
- Driver: UWP v30.0.13000.22008 (tested with 3 other previous official drivers and same issue)
- Wine version: N/A
- DXVK version: 1.9 (tested with previous 3 versions and still present).

Screenshots attached.
![126785305-eefbbc52-65bb-4a23-9759-ef593aa1870d](https://user-images.githubusercontent.com/87862595/126808039-2b6b8a04-6cc5-4f66-9f4f-ba6d6e1d68ed.jpg)
![126785310-5786472a-8fa8-4f5f-8c2e-5988731cb02f](https://user-images.githubusercontent.com/87862595/126808045-80b99560-e363-41e6-a2c7-e0ea7391d8de.jpg)

**Comment by K0bin** [comment]
Duplicate of #1862

**Comment by Fezzy976** [comment]
> Duplicate of #1862

WOW my issue is nowhere near that bad.

Is it possible to use the launch arguments from that post on windows or are they specific too linux?

these arguments
RADV_DEBUG=nodcc %command%
RADV_DEBUG=llvm GTA4EXE

**Comment by K0bin** [comment]
They are specific to Linux. Nothing you can do on Windows right now.

**Comment by Fezzy976** [comment]
> They are specific to Linux. Nothing you can do on Windows right now.

Thank you for the quick reply.

> > WOW my issue is nowhere near that bad.
> 
> It's the same problem though.
> 
> > Is it possible to use the launch arguments from that post on windows or are they specific too linux?
> 
> They are specific to Linux. Nothing you can do on Windows right now.

Ok thank you for the quick reply. What is the procedure for creating a log file on windows?

**Comment by K0bin** [comment]
No need for one. It wouldn't be useful anyway in this case and we also know what causes the problem. Don't expect a fix any time soon though.

**Comment by Fezzy976** [comment]
What would be the cause if you don’t mind? Curious to know.

**Comment by K0bin** [comment]
> What would be the cause if you don’t mind? Curious to know.

Rendering to and sampling from the same texture at the same time. That's allowed in D3D9 but isn't in Vulkan. Happens to just work on Nvidia hardware and DXVK has a workaround for AMD hardware but that doesn't work on RDNA2.

**Comment by stefson** [comment]
thanks for the warning about the rdn2 hardware, potentially saves me a ton of money! :+1:

**Comment by K0bin** [comment]
@stefson It's not a problem with the hardware, it's a problem with DXVK.

**Comment by stefson** [comment]
Sure, but DXVK is my only use case for such a high end card.

**Comment by K0bin** [comment]
And it works perfectly fine in all but like 3 games + this is probably gonna be solved in the future.

**Comment by Fezzy976** [comment]
I can confirm 100% that DXVK 1.7.3 does NOT have this issue!!!

It seems to be a bug within the newer version of DXVK, and not AMD drivers or hardware. I tested out 5K resolution using AMD VSR and with the three newest version of DXVK I tested all of them exhibt extreme flicker and screen corruption. The flicker is also present on lower resolutions but not that bad.

Now I tested DXVK 1.7.3 and the flicker is completely gone at 5K and all other resolutions too. Using the same drivers as before and it works flawlessly with this DXVK version.

**Comment by Squall-Leonhart** [comment]
not seeing an issue in 1.7.3 does not exclude hardware specific variation in driver or dxvk  behavior.

**Comment by adolfintel** [comment]
I have this glitch on linux using a 6900XT and RADV, DXVK 1.9.2

Can provide a trace if needed.

**Comment by K0bin** [comment]
Not necessary. We know what causes it but don't have any way to fix it.

**Comment by adolfintel** [comment]
I can confirm that 1.7.3 does not have this problem.
@K0bin were you able to bisect the exact commit that introduced the problem?

**Comment by Joshua-Ashton** [comment]
We know the fix, its just waiting on a Vulkan extension to do properly.

**Comment by bloudman648** [comment]
Had gfx issues with XClipse940 above 1.7.3 as well, uses RDNA Architecture. 
 
See here glitches newer dxvk versions although gou support up to vulkan 1.3 https://youtu.be/hORswuBVCEs

Can now play gta 4 natively on Android without gfx issues on 1080x2336
https://youtu.be/xc_SFCNYUXM
https://youtu.be/dfDrSEeLunE

**Comment by K0bin** [comment]
@bloudman648 We don't officially support Android.

Besides that, this is clearly a driver problem. Ask AMD to implement [VK_EXT_attachment_feedback_loop_layout](https://registry.khronos.org/vulkan/specs/latest/man/html/VK_EXT_attachment_feedback_loop_layout.html) in the Android Vulkan driver and then pray that the driver on your device will receive an update (it won't).

**Comment by bloudman648** [comment]
The App MiceWine : https://github.com/KreitinnSoftware/MiceWine-Application
 has integrated Mesa Wrapper Driver  + an Linux Environment with Wine (ESync), translates to DXVK, and as i said with DXVK 1.7.2 its working flawless.


The XClipse940 GPU is based on AMD RDNA3 architecture with supported ray tracing , although it's an mobile GPU.

Phone is Galaxy S24.

![Image](https://github.com/user-attachments/assets/9107e3aa-1de0-46b1-bab6-767c48fceb54)

**Comment by bloudman648** [comment]
![Image](https://github.com/user-attachments/assets/8a51281d-5bb8-48db-8200-1558e7bbac8e)

Vulkan 1.3 support...

**Comment by K0bin** [comment]
> The XClipse940 GPU is based on AMD RDNA3 architecture with supported hardware-accelerated ray tracing , (it's a phone GPU)

I know. I don't care what some stupid LLM says.

> has integrated Mesa Wrapper Driver

Irrelevant, it only has Turnip. RADV doesn't support Android.

> It wasn't a gfx driver issue, it was dxvk related just like the bug in the topic. (Desktop GPU)

The original issue was a driver problem. The video of your problem looks exactly like that original one. I don't know why older DXVK versions work for you but current DXVK doesn't have this issue on AMD GPUs with the Linux AMD driver or on Nvidia GPUs. I have to conclude that AMDs implementation of feedback loop layouts on xclipse or your particular driver is broken.

Either way, Android is not a supported platform for DXVK.

**Comment by Squall-Leonhart** [comment]
Doesn't older DXVK ask the driver to turn off DCC, which mitigates the problem by never being in a compressed layout.

**Comment by K0bin** [comment]
> Doesn't older DXVK ask the driver to turn off DCC, which mitigates the problem by never being in a compressed layout.

@Squall-Leonhart It did so using a RADV environment variable IIRC. So that wouldn't work for the proprietary AMD driver thats used on Android. I guess we just get lucky with the older DXVK version.

**Comment by Squall-Leonhart** [comment]
hrm, if they are using PAL on android, i recall seeing a variable to do the same, but it definitely wasn't named the same as RADV

**Comment by K0bin** [comment]
> so it relies on the universal VirGL driver. https://github.com/brunodev85/winlator

VirGL isn't really a driver. Instead it is used to pipe the Vulkan calls to the Vulkan driver of the Android device its running on.
